ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/preexecution/LogAppenderPreExecutingCommitStrategy.scala

```diff
@@ -21,16 +28,15 @@ class LogAppenderPreExecutingCommitStrategy(keySerializationStrategy: StateKeySe
       inputState: Map[DamlKvutils.DamlStateKey, Option[DamlKvutils.DamlStateValue]],
       preExecutionResult: KeyValueCommitting.PreExecutionResult,
   )(implicit executionContext: ExecutionContext)
-    : Future[PreExecutionCommitResult[RawKeyValuePairsWithLogEntry]] =
-    for {
-      serializedSuccessKeyValuePairs <- Future {
-        preExecutionResult.stateUpdates.map {
+    : Future[PreExecutionCommitResult[RawKeyValuePairsWithLogEntry]] = {
+    val serializedSuccessKeyValuePairs: SortedMap[Key, Value] =
```

Comment:
Please note that we'll be doing some CPU intensive operations outside of a `Future` here (i.e., compressing values).

Reply:
See above.